pom.xml

```diff
@@ -56,94 +65,124 @@
+        
+        <!-- Tests -->
+        
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13.2</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <pluginRepositories>
+        <pluginRepository>
+            <id>dash-licenses-snapshots</id>
+            <url>https://repo.eclipse.org/content/groups/releases/</url>
+        </pluginRepository>
+    </pluginRepositories>
+
     <build>
+        <resources>
+            <resource>
+                <directory>src/main/java</directory>
+                <includes>
+                    <include>**/*.properties</include>
+                    <include>**/*.xml</include>
+                </includes>
+            </resource>
+            <resource>
+                <directory>${project.basedir}/..</directory>
+                <includes>
+                    <include>LICENSE.md</include>
+                    <include>NOTICE.md</include>
+                </includes>
+                <targetPath>META-INF</targetPath>
+            </resource>
+            <resource>
+                <directory>src/main/resources</directory>
+            </resource>
+        </resources>
+        
         <plugins>
+            <!-- Sets minimal Maven version to 3.6.0 -->
             <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.0.0-M3</version>
                 <executions>
                     <execution>
-                        <id>enforce-versions</id>
+                        <id>enforce-maven</id>
                         <goals>
                             <goal>enforce</goal>
                         </goals>
                         <configuration>
                             <rules>
-                                <requireJavaVersion>
-                                    <version>[11,)</version>
-                                    <message>JDK11+ is required</message>
-                                </requireJavaVersion>
+                                <requireMavenVersion>
+                                    <version>3.6.0</version>
+                                </requireMavenVersion>
                             </rules>
                         </configuration>
                     </execution>
                 </executions>
             </plugin>
+            
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.8.1</version>
+                <version>3.11.0</version>
                 <executions>
                     <execution>
                         <id>default-compile</id>
                         <configuration>
-                            <release>${upper.java.level}</release>
+                            <release>11</release>
```

Comment:
I'd drop this line (to use corresponding property defined above).

Reply:
I'm not sure; it the number is used only once, isn't it better to use the constant as close as possible to where it's used?

I know there's a cult in Maven to make *everything* a variable, but I've never understand that cult. Besides, even the most hardcore fans of it never apply it consistently. In every pom that uses variables needlessly and excessively, there are always some things that could have been variables but aren't.

Reply:
Besides, `${upper.java.level}` is not used in other ee4j poms, so for consistency between projects alone I'd rather not use it (or change it in each and every ee4j project to be exactly that)

Reply:
It's the property for the compiler itself, not necessarily used to be referenced. But it can be removed again, since we already have the plugin. I'm also not sure whether this part is still needed:

```xml
<compilerArgs>
    <arg>-Xlint:all</arg>
    <arg>--add-reads</arg>
    <arg>org.glassfish.ha.api=ALL-UNNAMED</arg>
</compilerArgs>
```

It was in the existing pom.

Reply:
@pzygielo based on your comments and that of the others I've simplified matters. Sorry for the initial confusion. Can you take a look again?